src/OrchardCore/OrchardCore/Shell/ShellHost.cs

```diff
@@ -144,7 +144,6 @@ public async Task<ShellScope> GetScopeAsync(ShellSettings settings)
         public async Task UpdateShellSettingsAsync(ShellSettings settings)
         {
             settings.VersionId = IdGenerator.GenerateId();
-            settings.TenantId ??= IdGenerator.GenerateId();
```

Comment:
Here the central place I was thinking about ;)

Reply:
This method is called UpdateShellSettingsAsync so its function is to update the ShellSettings and here it can mutate the TenantId if someone removes the TenantId from the tenant.json file. This is why I removed it from here. This method is surely called in a lot of custom modules to change the ShellSettings for their own need with their extensions. This means that if they don't have a TenantId it will append one to the tenant.json file but I would rather throw an exception if a TenantId is missing instead of having a method that could potentially morph the TenantId.

Reply:
We should update the existing loaded (or 1st created) settings otherwise it is not consistent for any other property. As we do for content items, document manager, the pattern is to cache immutable instance and load mutable instances for updating.